proposals/0448-type-variable-scoping.rst

```diff
@@ -408,6 +414,13 @@ Proposed Change Specification
        tyapp_or_pat → '@' atype    -- '@' is in prefix position
                     → apat
 
+#. For backwards compatibility, accept the previous extension to the pattern grammar (a) if ``-XTypeAbstractions`` is enabled or (b) if both ``-XScopedTypeVariables`` and ``-XTypeApplications`` are enabled.
```

Comment:
This is still confusing.  What is "the previous extension".  Do you mean previous to this proposal?  No, you mean the thing in item (1)!

Also we don't need (a) because it is already covered in item (1), and that is nothing to do with back-compat.  Try this
```
3. For backward compatiblity, *also* accept type application syntax in constructor patterns if both
   ``-XScopedTypeVariables`` and ``-XTypeApplications`` are enabled, but ``-XTypeAbstractions`` is not.
    In that case, emit a warning, stating that type applications in constructor patterns should be
    enabled with ``-XTypeAbstractions``, and that the temporary expedient of enabling it by the 
   combination of ``-XScopedTypeVariables`` and ``-XTypeApplications`` will be removed.

   After 2 releases remove clause (b); ``-XTypeAbstractions`` will be the only way to enable this feature.
```

Reply:
I applied this manually, thanks!